src/apps/lwaftr/lwaftr.lua

```diff
@@ -254,16 +254,48 @@ local function in_binding_table(lwstate, ipv6_src_ip, ipv6_dst_ip, ipv4_src_ip,
    return b4 and ipv6_equals(b4, ipv6_src_ip) and ipv6_equals(br, ipv6_dst_ip)
 end
 
+local function transmit_ipv4(lwstate, pkt)
+   local ipv4_header = get_ethernet_payload(pkt)
+   local dst_ip = get_ipv4_dst_address(ipv4_header)
+   if lwstate.hairpinning and ipv4_in_binding_table(lwstate, dst_ip) then
+      -- The destination address is managed by the lwAFTR, so we need to
+      -- hairpin this packet.  Enqueue on the IPv4 interface, as if it
+      -- came from the internet.
+      return transmit(lwstate.input.v4, pkt)
+   else
+      return transmit(lwstate.o4, pkt)
+   end
+end
+
+local function transmit_ipv4_reply(lwstate, pkt, orig_pkt)
+   drop(orig_pkt)
+   return transmit_ipv4(lwstate, pkt)
+end
+
 -- ICMPv4 type 3 code 1, as per RFC 7596.
 -- The target IPv4 address + port is not in the table.
-local function icmp_after_discard(lwstate, pkt, to_ip)
-   local icmp_config = {type = constants.icmpv4_dst_unreachable,
-                        code = constants.icmpv4_host_unreachable,
-                        }
-   local icmp_dis = icmp.new_icmpv4_packet(lwstate.aftr_mac_inet_side, lwstate.inet_mac,
-                                           lwstate.aftr_ipv4_ip, to_ip, pkt,
-                                           ethernet_header_size, icmp_config)
-   return transmit(lwstate.o4, icmp_dis)
+local function drop_ipv4_packet_to_unreachable_host(lwstate, pkt, to_ip)
+   if lwstate.policy_icmpv4_outgoing == lwconf.policies['DROP'] then
+      -- ICMP error messages off by policy; silently drop.
+      return drop(pkt)
+   end
+
+   if get_ipv4_proto(get_ethernet_payload(pkt)) == proto_icmp then
+      -- FIXME: We should send dst/host unreachable for ICMP
+      -- messages, unless the ICMP message is an error message.
```

Comment:
Following the discussion in #263, I guess I should change this FIXME to refer to the RFC.